master-detail-demo/src_hd/com/axonivy/demo/masterdetail/ProductList/ProductDetail.xhtml

```diff
@@ -0,0 +1,123 @@
+						</div>
+					</div>
+					
+					<div class="ui-g">
+						<div class="ui-g-4">
+							<p:outputLabel for="orderDate" value="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/orderDate')}" />
+						</div>
+						<div class="ui-g-4">
+							<p:datePicker id="orderDate" value="#{data.product.orderDate}" pattern="dd.MM.yyyy"
+									required="true" requiredMessage="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/required')} #{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/orderDate')}" />
+						</div>
+					</div>
+					
+					<div class="ui-g">
+						<div class="ui-g-4">
+							<p:outputLabel for="deliveryDate" value="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/deliveryDate')}" />
+						</div>
+						<div class="ui-g-4">
+							<p:datePicker id="deliveryDate" value="#{data.product.deliveryDate}" pattern="dd.MM.yyyy"
+									required="true" requiredMessage="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/required')} #{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/deliveryDate')}" />
+						</div>
+					</div>	
+					
+					<div class="ui-g">
+						<div class="ui-g-4">
+							<p:outputLabel for="quantity" value="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/quantity')}" />
+						</div>
+						<div class="ui-g-4">
+							<p:inputNumber id="quantity" value="#{data.product.quantity}" decimalPlaces="0"
+									required="true" requiredMessage="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/required')} #{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/quantity')}" />
+						</div>
+					</div>
+					
+					<div class="ui-g">
+						<div class="ui-g-4">
+							<p:outputLabel for="quality" value="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/quality')}" />
+						</div>
+						<div class="ui-g-4">
+							<p:selectOneMenu id="quality" value="#{data.product.quality}" converter="pojoConverter" 
+									required="true" requiredMessage="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/required')} #{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/quality')}">
+								<f:selectItem noSelectionOption="true" />
+								<f:selectItems value="#{enumBean.qualityValues}" var="quality" itemLabel="#{quality.getCmsName()}" />
+							</p:selectOneMenu>
+						</div>
+					</div>
+					
+					<div class="ui-g">
+						<div class="ui-g-4">
+							<p:outputLabel for="onSale" value="#{ivy.cms.co('/Dialogs/com/axonivy/demo/statefuldatatable/ui/StatefulDatatable/onSale')}" />

```

Comment:
@nqhoan-axonivy  do you think this view is safe?

Reply:
yep. looks good to me